### ISSUE TITLE
False positive for SC2034 when var used in array parameter expansion as offset or length

### ISSUE THREAD

**Issue by wedge-jarrad** [koalaman/shellcheck#1124]
- SC2034: 
- My shellcheck version ("online"): 
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet that shows the problem:

```sh
#!/bin/bash
arr=(1 2 3 4)
num=2
echo "${arr[@]:0:num}"
echo "${arr[@]:num:1}"
echo "${arr[@]:num}"
```

#### Here's what shellcheck currently says:
```
Line 3:
num=2
^-- SC2034: num appears unused. Verify it or export it.
```

**Comment by koalaman** [comment]
Fixed by @ngzhian. Thanks!
